src/WebJobs.Extensions.DurableTask/EventGridLifeCycleNotificationHelper.cs

```diff
@@ -42,91 +42,129 @@ public EventGridLifeCycleNotificationHelper(
+                    this.ManagedIdentityTokenSource = new ManagedIdentityTokenSource("https://eventgrid.azure.net/.default");
+                }
+            }
+            else
+            {
+                eventGridNotificationsConfig = options.Notifications.EventGrid ?? throw new ArgumentNullException(nameof(options.Notifications.EventGrid));
+
+                this.eventGridKeyValue = nameResolver.Resolve(eventGridNotificationsConfig.KeySettingName);
+
+                this.eventGridTopicEndpoint = eventGridNotificationsConfig.TopicEndpoint;
+
+                if (nameResolver.TryResolveWholeString(eventGridNotificationsConfig.TopicEndpoint, out var endpoint))
+                {
+                    this.eventGridTopicEndpoint = endpoint;
+                }
+            }
+
+            // Check if we have the minimum required settings to enable Event Grid notifications with key based authentication.
+            bool eventGridNotificationSettingsConfigured = false;
+
+            if (eventGridNotificationsConfig != null)
+            {
+                if (!string.IsNullOrEmpty(eventGridNotificationsConfig.KeySettingName) && string.IsNullOrEmpty(this.eventGridKeyValue))
+                {
+                    throw new ArgumentException($"Failed to start lifecycle notification feature. Please check the configuration values for {eventGridNotificationsConfig.KeySettingName} on AppSettings.");
+                }
+
+                if (string.IsNullOrEmpty(eventGridNotificationsConfig.KeySettingName) || string.IsNullOrEmpty(eventGridNotificationsConfig.TopicEndpoint))
+                {
+                    throw new ArgumentException($"Failed to start lifecycle notification feature. Please check the configuration values for {eventGridNotificationsConfig.TopicEndpoint} and {eventGridNotificationsConfig.KeySettingName}.");
+                }
+
+                eventGridNotificationSettingsConfigured = true;
+            }
 
-            if (!string.IsNullOrEmpty(this.eventGridTopicEndpoint))
-            {
-                if (!string.IsNullOrEmpty(eventGridNotificationsConfig.KeySettingName))
-                {
-                    this.useTrace = true;
+            if (this.UseManagedIdentity || eventGridNotificationSettingsConfigured)
+            {
+                this.useTrace = true;
 
-                    var retryStatusCode = eventGridNotificationsConfig.PublishRetryHttpStatus?
-                                              .Where(x => Enum.IsDefined(typeof(HttpStatusCode), x))
-                                              .Select(x => (HttpStatusCode)x)
-                                              .ToArray()
-                                          ?? Array.Empty<HttpStatusCode>();
-
-                    if (eventGridNotificationsConfig.PublishEventTypes == null || eventGridNotificationsConfig.PublishEventTypes.Length == 0)
+                var retryStatusCode = eventGridNotificationsConfig != null ?
```

Comment:
It looks like the previous logic was setting the `retryStatusCode` to `Array.Empty` in the case that `eventGridNotificationsConfig.PublishRetryHttpStatus` is null, but I think the current logic will instead set the `retryStatusCode` to null in this situation (and set it to `Array.Empty` only if the entire `eventGridNotificationsConfig` is null). I think it should be 
```suggestion
                var retryStatusCode = eventGridNotificationsConfig?.PublishRetryHttpStatus != null ?
```
That way, if `eventGridNotificationsConfig` itself is null, we will set `retryStatusCode` to `Array.Empty`, and will also do the same if the `PublishRetryHttpStatus` field is.